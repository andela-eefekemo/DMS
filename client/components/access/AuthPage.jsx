import React, { Component } from 'react';
import Header from '../include/Header';

/**
 * @class LandingContainer
 * @extends {Component}
 */
class AuthPage extends Component {
  constructor(props) {
    super(props);
  }
  /**
   * @returns {jsx} -
   * @memberof LandingContainer
   */
  render() {
    return (
      <div>
        <Header match={this.props.match} />
        <div className="container authpage">
          <div className="row auth-card">
            {this.props.children}
          </div>
        </div>
      </div>
    );
  }
}

export default AuthPage;
