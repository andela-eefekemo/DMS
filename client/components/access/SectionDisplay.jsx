import React from 'react';
import { Link } from 'react-router-dom';
/**
 * Section Display component that displays a carousel.
 * @returns {String} The HTML markup for the Section Display component
 */
const SectionDisplay = () => (
  <div className="section">
    <div className="carousel carousel-slider center" data-indicators="true">
      <div className="carousel-fixed-item">
        <span className="inner-slide">
          <img
            src="../../img/slide 3.jpg"
            alt="smallslide2" className="slide-img" />
        </span>
        <div className="landing-content">
          <h2>First Panel</h2>
          <Link to="/signup">Get Started</Link>
        </div>
      </div>
      <div className="carousel-item white-text" href="#one!">
        <img
          src="../../img/smallslide2.jpg"
          alt="smallslide2" className="slide-img" />
      </div>
      <div className="carousel-item amber white-text" href="#two!">
        <img
          src="../../img/smallslide2.jpg"
          alt="smallslide2" className="slide-img" />
      </div>
      <div className="carousel-item green white-text" href="#three!">
        <img
          src="../../img/smallslide2.jpg"
          alt="smallslide2" className="slide-img" />
      </div>
      <div className="carousel-item blue white-text" href="#four!">
        <img
          src="../../img/smallslide2.jpg"
          alt="smallslide2" className="slide-img" />
      </div>
    </div>
  </div>
);

export default SectionDisplay;
