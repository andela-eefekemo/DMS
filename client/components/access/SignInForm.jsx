import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import AccessActions from '../../actions/AccessActions';
import InputField from '../common/InputField';
import validate from '../../utilities/validate';

const signInUser = AccessActions.signInUser;
/**
 * @class SignInForm
 * @extends {Component}
 */
class SignInForm extends Component {
  /**
 * Creates an instance of SignUpForm.
 * @param {any} props -
 * @memberof SignInForm
 */
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: ''
    };
    this.onSubmit = this.onSubmit.bind(this);
    this.onChange = this.onChange.bind(this);
  }

  /**
   * @return {void}
   * @param {any} e -
   * @memberof SignInForm
   */
  onSubmit(e) {
    e.preventDefault();
    try {
      if (!validate(this.state)) {
        throw new Error('No field should be left blank');
      }
      this.props.signInUser(this.state)
        .then(() => {
          Materialize.toast('Success!', 2000, 'indigo darken-4 white-text rounded');
          this.context.router.history.push('/dashboard');
        });
    } catch (err) {
      Materialize.toast(err.message, 3000,
        'indigo darken-4 white-text rounded');
    }
  }

  /**
   * @return {void}
   * @param {any} e -
   * @memberof SignInForm
   */
  onChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }
  /**
   * @returns {jsx} -
   * @memberof SignInForm
   */
  render() {
    return (
      <form onSubmit={this.onSubmit}>
        <h5 className="center"> Sign In!</h5>
        <InputField
          name="email"
          label="Email"
          className="validate form-design"
          type="email" onChange={this.onChange} />
        <InputField
          name="password"
          label="Password"
          className="validate form-design"
          type="password" onChange={this.onChange} />
        <div className="input-field center">
          <button className="waves-effect btn button-design" type="submit">
            Sign In
          </button>
        </div>
      </form>
    );
  }
}

SignInForm.propTypes = {
  signInUser: PropTypes.func.isRequired
};

SignInForm.contextTypes = {
  router: PropTypes.object.isRequired
};

export default connect(null, { signInUser })(SignInForm);
