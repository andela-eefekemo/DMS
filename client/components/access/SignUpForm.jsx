import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import AccessActions from '../../actions/AccessActions';
import InputField from '../common/InputField';
import validate from '../../utilities/validate';


const signUpUser = AccessActions.signUpUser;
/**
 * @class SignUpForm
 * @extends {Component}
 */
class SignUpForm extends Component {
  /**
   * Creates an instance of SignUpForm.
   * @param {any} props
   * @param {any} context
   * @memberof SignUpForm
   */
  constructor(props, context) {
    super(props, context);
    this.state = {
      firstName: '',
      lastName: '',
      email: '',
      password: '',
      confirmPassword: ''
    };
    this.onSubmit = this.onSubmit.bind(this);
    this.onChange = this.onChange.bind(this);
  }

  /**
   * @return {void}
   * @param {any} e -
   * @memberof SignUpForm
   */
  onSubmit(e) {
    e.preventDefault();
    try {
      if (!validate(this.state)) {
        throw new Error('No field should be left blank');
      }
      this.props.signUpUser(this.state)
        .then(() => {
          debugger;
          Materialize.toast('Success!', 2000, 'indigo darken-4 white-text rounded');
          this.context.router.history.push('/dashboard');
        });
    } catch (err) {
      Materialize.toast(err.message, 3000,
        'indigo darken-4 white-text rounded');
    }
  }

  /**
   * @return {void}
   * @param {any} e -
   * @memberof SignUpForm
   */
  onChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  /**
   * @returns {jsx} -
   * @memberof SignUpForm
   */
  render() {
    return (
      <form onSubmit={this.onSubmit}>
        <h5 className="center"> Create an account </h5>
        <div className="row">
          <div className="col l6 m6 s12">
            <InputField
              name="firstName"
              value={this.state.firstName}
              placeholder="First Name"
              className="validate form-design"
              type="text" onChange={this.onChange} />
          </div>
          <div className="col l6 m6 s12">
            <InputField
              name="lastName"
              value={this.state.lastName}
              placeholder="Last Name"
              className="validate form-design"
              type="text" onChange={this.onChange} />
          </div>
          <div className="col l12 m12 s12">
            <InputField
              name="email"
              value={this.state.email}
              placeholder="Email"
              className="validate form-design"
              type="email" onChange={this.onChange} />
          </div>
          <div className="col l12 m12 s12">
            <InputField
              name="password"
              value={this.state.password}
              placeholder="Password"
              className="validate form-design"
              type="password" onChange={this.onChange} />
          </div>
          <div className="col l12 m12 s12">
            <InputField
              name="confirmPassword"
              value={this.state.confirmPassword}
              placeholder="Re-Type Password"
              className="validate form-design"
              type="password" onChange={this.onChange} />
          </div>
        </div>
        <div className="input-field center">
          <button className="waves-effect btn button-design" type="submit">
            Sign Up
          </button>
        </div>
        <p className="center">Have an account? <Link to="signin">Sign In</Link></p>
      </form>
    );
  }
}

SignUpForm.propTypes = {
  signUpUser: PropTypes.func.isRequired
};

SignUpForm.contextTypes = {
  router: PropTypes.object.isRequired
};

export default connect(null, { signUpUser })(SignUpForm);
