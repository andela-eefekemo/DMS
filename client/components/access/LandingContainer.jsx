import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ConnectedHeader from '../include/Header';
import SectionDisplay from './SectionDisplay';

/**
 * @class LandingContainer
 * @extends {Component}
 */
export class LandingContainer extends Component {

  /**
   * @return {void}
   * @memberof LandingContainer
   */
  componentDidMount() {
    $('.carousel.carousel-slider').carousel({ fullWidth: true });
    if (this.props.access.isAuthenticated === true) {
      this.props.history.push('/dashboard');
    }
  }

  /**
   * @returns {jsx} -
   * @memberof LandingContainer
   */
  render() {
    return (
      <div className="landing-page">
        <ConnectedHeader match={this.props.match} />
        <SectionDisplay />
      </div>
    );
  }
}

const mapPropsToState = (state) => {
  return {
    access: state.access
  };
};

LandingContainer.propTypes = {
  access: PropTypes.object.isRequired,
  match: PropTypes.object.isRequired,
  history: PropTypes.object
};


export default connect(mapPropsToState)(withRouter(LandingContainer));
