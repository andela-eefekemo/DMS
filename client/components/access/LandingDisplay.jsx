import React from 'react';

import SignInForm from './SignInForm';
import SignUpForm from './SignUpForm';

const LandingDisplay = () => (
  <div className="row auth-card">
    <div className="card">
      <div className="card-tabs">
        <ul className="tabs tabs-fixed-width">
          <li className="tab">
            <a className="active" href="#signUp">Sign Up</a>
          </li>
          <li className="tab">
            <a href="#signIn">Sign In</a>
          </li>
        </ul>
      </div>
      <div className="card-content">
        <div id="signUp">
          <SignUpForm />
        </div>
        <div id="signIn">
          <SignInForm />
        </div>
      </div>
    </div>
  </div>
);

export default LandingDisplay;
