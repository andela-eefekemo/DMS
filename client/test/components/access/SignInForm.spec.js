/* global expect jest test */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { SignInForm } from '../../../components/access/SignInForm';

describe('SignInForm Component', () => {
  const spy = jest.fn();
  beforeEach(() => {
    global.Materialize = { toast: jest.fn() };
  });
  afterEach(() => {
    spy.mockReset();
  });
  const props = {
    match: {
      path: ''
    },
    access: {
      isAuthenticated: false
    },
    history: {
      push: spy
    },
    signInUser: jest.fn(() => {
      return Promise.resolve();
    })
  };
  const component = shallow(
    <SignInForm {...props} />);
  test('should match the SignInForm snapshot', () => {
    const tree = toJson(component);
    expect(component.find('input').length).toEqual(2);
    expect(tree).toMatchSnapshot();
  });
  test('it should set state when onChange function is called', () => {
    component.instance().onChange(
      { target: { value: 'efe@gmail.com', name: 'email' } });
    component.instance().onChange(
      { target: { value: 'eguono', name: 'password' } });
    expect(component.state('email')).toEqual('efe@gmail.com');
    expect(component.state('password')).toEqual('eguono');
  });
  test(
    'it should submit fields in state when onSubmit function is called', () => {
      jest.spyOn(component.instance(), 'onSubmit');
      component.instance().onChange(
        { target: { value: 'efe@gmail.com', name: 'email' } });
      component.instance().onChange(
        { target: { value: 'eguono', name: 'password' } });
      component.find('#submit-signin').simulate('click');
      expect(component.find('#submit-signin').length).toEqual(1);
      expect(component.instance().onSubmit.mock.calls.length).toEqual(1);
    });
  test('onSubmit function should run', () => {
    component.instance().onChange(
      { target: { value: 'eguono@gmail.com', name: 'email' } });
    component.instance().onChange(
      { target: { value: 'efe', name: 'password' } });
    expect(component.state('email')).toEqual('eguono@gmail.com');
    expect(component.state('password')).toEqual('efe');
    const newspy = jest.spyOn(component.instance(), 'onSubmit');
    component.instance().onSubmit();
    expect(newspy).toHaveBeenCalled();
  });
});
