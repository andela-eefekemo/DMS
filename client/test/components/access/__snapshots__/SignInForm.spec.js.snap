// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInForm Component should match the SignInForm snapshot 1`] = `
<div>
  <h5
    className="center"
  >
     Log Into Doc-ms
  </h5>
  <div
    className="input-field"
  >
    <input
      className="validate form-design"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      required={true}
      type="email"
    />
  </div>
  <div
    className="input-field"
  >
    <input
      className="validate form-design"
      name="password"
      onChange={[Function]}
      placeholder="password"
      required={true}
      type="password"
    />
  </div>
  <div
    className="input-field center"
  >
    <button
      className="waves-effect btn button-design"
      id="submit-signin"
      onClick={[Function]}
    >
      Sign In
    </button>
  </div>
  <p
    className="center"
  >
    Don't Have an account? 
    <Link
      replace={false}
      to="signup"
    >
      Create an Account
    </Link>
  </p>
</div>
`;
